backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/cgen/CBridgeGen.kt

```diff
@@ -936,7 +936,9 @@ private class StructValuePassing(private val kotlinClass: IrClass, override val
         bridgeCallBuilder.prepare += kotlinPointed
 
         val cPointer = passThroughBridge(irGet(kotlinPointed), kotlinPointedType, CTypes.pointer(cType))
-        cBridgeBodyLines += "*${cPointer.name} = $expression;"
+        val kniStructResult = "kniStructResult"
+        cBridgeBodyLines += "${cType.render(kniStructResult)} = $expression;"
+        cBridgeBodyLines += "memcpy(${cPointer.name}, &$kniStructResult, sizeof($kniStructResult));"
```

Comment:
What is this required for?

Reply:
+1, tailcall seems to be enough (and more efficient).

Reply:
It is required for structs with `const` fields. It was done like this [before].(https://github.com/JetBrains/kotlin-native/commit/471771e8b8c7e6e63b72df16538bc034af433c56).

Reply:
CBridgeGen shouldn't get structs with const fields. All structs it manipulates are generated by cinterop.

Have you fixed any particular reproducible issue with this patch?

Reply:
https://github.com/JetBrains/kotlin-native/blob/master/Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StructRendering.kt

Reply:
Yes, i've added these lines because of [this](https://github.com/JetBrains/kotlin-native/blob/master/backend.native/tests/interop/basics/types.kt#L6) failing test.

Reply:
It turns out that the logic I referred above contains a bug:
https://github.com/JetBrains/kotlin-native/blob/712c056a547b0da1ea7dfca38e3b452d75e0bff7/Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StructRendering.kt#L67
This actually can place `const` modifier.

This is not supposed to happen (e.g. to simplify CBridgeGen), so the proper way to fix this is to adjust cinterop (most likely fix `type.spelling` or don't rely on it in struct rendering).

Reply:
> the proper way to fix this is to adjust cinterop (most likely fix type.spelling or don't rely on it in struct rendering).

Is there any progress on this?